libr/core/cmd_search.c

```diff
@@ -629,29 +629,50 @@ R_API RList *r_core_get_boundaries_prot(RCore *core, int protection, const char
 		if (m) {
 			append_bound (list, core->io, search_itv, m->itv.addr, m->itv.size);
 		}
-	} else if (!strcmp (mode, "io.maps")) { // Non-overlapping RIOMap parts not overriden by others (skyline)
+	} else if (r_str_startswith (mode, "io.maps")) { // Non-overlapping RIOMap parts not overriden by others (skyline)
 		const RPVector *skyline = &core->io->map_skyline;
 		ut64 begin = UT64_MAX;
 		ut64 end = UT64_MAX;
 		size_t i;
+		int mask = 0;
+		bool readonly = false;
+
+		if (!strcmp (mode, "io.maps.exec")) {
+			mask |= R_IO_EXEC;
+		}
+		if (!strcmp (mode, "io.maps.write")) {
+			mask |= R_IO_WRITE;
+		}
+		if (!strcmp (mode, "io.maps.readonly")) {
+			readonly = true;
+		}
+
 		for (i = 0; i < r_pvector_len (skyline); i++) {
```

Comment:
if skyline is going to die or change this code will need to be changed, im not sure whats the best way to walk into overlapped io maps. cc @MaskRay @condret

Reply:
skyline and submaps are the same thing, sorted array of un-shadowed IO maps.

Their construction is different:

* skyline: O(n log n) rebuild but batch insertion is supported
* submaps: O(n) which also relies on r_oids* stuff. The overall complexity is higher (you also have to learn idpool stuff). The 

I don't see why "skyline is going to die". I have strong opinion regarding the implementation of that submaps PR.